### PR TITLE
check expiry of root during checkRoot

### PR DESCRIPTION
**Pull request by endophage**
@NathanMcCauley @diogomonica

### CODE REVIEW

**Comment by diogomonica** [comment]
I'm not waiting for mccauley: Tests?!

File: client/client.go [hunk 1]
```diff
@@ -114,6 +114,16 @@ func (c Client) checkRoot() error {
 	if !bytes.Equal(hash[:], hashSha256) {
 		return fmt.Errorf("Cached root sha256 did not match snapshot root sha256")
 	}
+
+	root := &data.SignedRoot{}
+	err = json.Unmarshal(raw, root)
+	if err != nil {
+		return ErrCorruptedCache{file: "root.json"}
```
Comment (diogomonica) [inline client/client.go:125]: Is this being caught and `file` being used anywhere? I guess the question is why do we have ErrCorruptedCache. Should we do something different with it?

Comment (endophage) [inline client/client.go:125]: All errors out of checkRoot get converted into ErrLocalRootExpired in the `update()` function above.

This should already be covered by the client tests but apparently we don't have any that include local expired data. Need to look at adding to them.

**Comment by NathanMcCauley** [comment]
LGTM

**Pull request merged**